modules/core/src/main/java/org/lwjgl/system/MemoryUtil.java

```diff
@@ -1766,4 +1766,105 @@ public static String memUTF16(ByteBuffer buffer, int length, int offset) {
 	}
 
-}
\ No newline at end of file
+	/**
+	 * Uses the vsnprintf() C library function to format a null-terminated string with a {@code va_list} argument list, then converts the result
+	 * to an ASCII encoded {@link String}.
+	 *
+	 * <p>The temporary storage required to format the string is allocated on the stack. If the needed storage size surpasses the allowed capacity, the
+	 * result is truncated.</p>
+	 *
+	 * @param stack        {@link MemoryStack} to buffer the formatted result
+	 * @param capacity     the stack size allowed to be used
+	 * @param formatString memory address of the format string
+	 * @param valist       memory address of the {@code va_list} argument list
+	 *
+	 * @return the decoded {@link String}
+	 */
+	public static String memASCIIVPrintf(MemoryStack stack, int capacity, long formatString, long valist) {
+		long address = stack.nmalloc(capacity);
+		ACCESSOR.vsnPrintf(address, capacity, formatString, valist);
+		return memASCII(address);
+	}
+
+	/**
+	 * Uses the vsnprintf() C library function to format a null-terminated string with a {@code va_list} argument list, then converts the result
+	 * to a UTF-8 encoded {@link String}.
+	 *
+	 * <p>The temporary storage required to format the string is allocated on the stack. If the needed storage size surpasses the allowed capacity, the
+	 * result is truncated.</p>
+	 *
+	 * @param stack        {@link MemoryStack} to buffer the formatted result
+	 * @param capacity     the stack size allowed to be used
+	 * @param formatString memory address of the format string
+	 * @param valist       memory address of the {@code va_list} argument list
+	 *
+	 * @return the decoded {@link String}
+	 */
+	public static String memUTF8VPrintf(MemoryStack stack, int capacity, long formatString, long valist) {
+		long address = stack.nmalloc(capacity);
+		ACCESSOR.vsnPrintf(address, capacity, formatString, valist);
+		return memUTF8(address);
+	}
+
+	/**
+	 * Uses the vsnprintf() C library function to format a null-terminated string with a {@code va_list} argument list. The result is stored at index
+	 * {@code [0, buffer.remaining()}) in {@code buffer}, as a null-terminated string.
+	 *
+	 * <p>Output is truncated if there is not enough space in {@code [0, buffer.remaining()}). In this case, this function returns the size which would
+	 * have been required to store the full string, excluding the null terminator.</p>
+	 *
+	 * <p>The current {@code position} and {@code limit} of the specified {@code buffer} are not affected by this operation.</p>
+	 *
+	 * @param buffer       the output buffer
+	 * @param formatString memory address of the format string
+	 * @param valist       memory address of the {@code va_list} argument list
+	 *
+	 * @return the number of characters written to {@code buffer}, excluding the null terminator, or a negative value in case of error
+	 */
+	public static int memVPrintf(ByteBuffer buffer, long formatString, long valist) {
```

Comment:
I think this is the only method we should keep. It will be used very rarely and it's OK to write some extra code to handle the other overloads.